conans/test/unittests/client/tools/system_pm_test.py

```diff
@@ -526,3 +517,144 @@ def get_linux_error_message():
+            spt = SystemPackageTool(runner=runner, tool=AptTool(output=self.out), output=self.out)
+            spt.install_packages(packages)
+            self.assertEqual(5, runner.calls)
+
+    def test_installed_packages_and_install_variant(self):
+        """ Only variant must be installed. Packages are already installed
+        """
+        packages = [("varianta", "variantb", "variantc"), "a_package", "another_package"]
+        with tools.environment_append({"CONAN_SYSREQUIRES_SUDO": "True"}):
+            runner = RunnerMultipleMock(['dpkg-query -W -f=\'${Status}\' a_package | '
+                                         'grep -q "ok installed"',
+                                         'dpkg-query -W -f=\'${Status}\' another_package | '
+                                         'grep -q "ok installed"',
+                                         "sudo -A apt-get update",
+                                         "sudo -A apt-get install -y --no-install-recommends"
+                                         " varianta",
+                                         ])
+            spt = SystemPackageTool(runner=runner, tool=AptTool(output=self.out), output=self.out)
+            spt.install_packages(packages)
+            self.assertEqual(7, runner.calls)
+
+    def test_variants_and_packages_installed(self):
+        """ Install nothing, all is already installed
+        """
+        packages = [("varianta", "variantb", "variantc"), "a_package", "another_package"]
+        with tools.environment_append({"CONAN_SYSREQUIRES_SUDO": "True"}):
+            runner = RunnerMultipleMock(['dpkg-query -W -f=\'${Status}\' varianta | '
+                                         'grep -q "ok installed"',
+                                         'dpkg-query -W -f=\'${Status}\' a_package | '
+                                         'grep -q "ok installed"',
+                                         'dpkg-query -W -f=\'${Status}\' another_package | '
+                                         'grep -q "ok installed"',
+                                         ])
+            spt = SystemPackageTool(runner=runner, tool=AptTool(output=self.out), output=self.out)
+            spt.install_packages(packages)
+            self.assertEqual(3, runner.calls)
+
+    def test_empty_variants_and_packages(self):
+        packages = [(),]
+        with tools.environment_append({"CONAN_SYSREQUIRES_SUDO": "True"}):
+            runner = RunnerMultipleMock()
+            spt = SystemPackageTool(runner=runner, tool=AptTool(output=self.out),
+                                    output=self.out)
+            spt.install_packages(packages)
+            self.assertEqual(0, runner.calls)
+
+    def test_install_all_multiple_package_list(self):
+        """ Separated string list must be treated as full package list to be installed
+        """
+        packages = "varianta variantb", "variantc variantd"
+        with tools.environment_append({"CONAN_SYSREQUIRES_SUDO": "True"}):
```

Comment:
Is this needed here?

Reply:
No, it's a copy-past error. gonna fix it.